kubeflow/trainer/rhai/transformers.py

```
@@ -399,9 +398,20 @@ def __init__(self, cloud_remote_storage_uri: Optional[str] = None) -> None:
                         test_file = ".kubeflow-access-test"
-                        self.remote_fs.pipe(test_file, b"test")
-                        self.remote_fs.cat(test_file)
-                        self.remote_fs.rm_file(test_file)
+                        last_error = None
+                        for attempt in range(1, 4):  # 3 attempts with backoff
+                            try:
+                                self.remote_fs.pipe(test_file, b"test")
+                                self.remote_fs.cat(test_file)
+                                self.remote_fs.rm_file(test_file)
+                                break  # Success
+                            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)